nixui/nix_eval.py

```diff
@@ -24,22 +24,41 @@ def nix_instantiate_eval(expr, strict=False):
 
 def get_modules_defined_attrs(module_path, attr_loc=[]):
     attributes = {}
-    for sub_attr_datapoint in get_subattribute_data(module_path, attr_loc):
-        full_attr_loc = attr_loc + [sub_attr_datapoint['name']]
-        if sub_attr_datapoint['position'] is not None and sub_attr_datapoint['position']['file'] == module_path:
-            if sub_attr_datapoint['type'] != 'set':
-                attributes[tuple(full_attr_loc)] = sub_attr_datapoint
-            else:
-                res = get_modules_defined_attrs(
-                    module_path,
-                    full_attr_loc
-                )
-                if not res:
-                    attributes[tuple(full_attr_loc)] = sub_attr_datapoint
-                else:
-                    attributes.update(res)
-    return attributes
 
+    return {
+        tuple(v['name']): {"position": v['position']}
+        for v in nix_instantiate_eval("""
```

Comment:
nit: we shouldn't have huge strings within generators.

```
result = nix_instantiate_eval("""
...
...
""")
return {tuple(v['name']): {"position": v['position']} for v in result}
```

for

Reply:
See a3bfbd0889c92dd01b91eaf3c160abe7a04d3e33